src/main/java/io/rcktapp/api/handler/firehose/FirehoseDb.java

```diff
@@ -97,6 +101,25 @@ public void bootstrapApi() throws Exception {
         }
     }
 
+    private List<String> listAllDeliveryStreamNames() {
+        ListDeliveryStreamsRequest listDeliveryStreamsRequest = new ListDeliveryStreamsRequest().withDeliveryStreamType("DirectPut");
+        ListDeliveryStreamsResult listDeliveryStreamsResult;
+        List<String> deliveryStreamNames = new ArrayList<>();
+
+        do {
+            if (!deliveryStreamNames.isEmpty()) {
```

Comment:
Do you understand the reduce operation I used to get the ultimate entry from a list without using all this conditional /mathing?  You're still using a magic number.

Reply:
Also, I think this should happen just before the while - not way up here at the top of the loop.  you're *always* going to have this isEmpty on the first path.  Why not wait for a chance for it to have a value before you start mucking with it?  (It might be empty anyway if there are no stream names, too)

Reply:
I think this should happen just before the while - don't see how makes any sense to have it up here at the top of the request, mutating the request.